bika/lims/browser/fields/historyawarereferencefield.py

```diff
@@ -78,19 +76,10 @@ def set(self, instance, value, **kwargs):
         for uid in newuids:
             # update version_id of all existing references that aren't
             # about to be removed anyway (contents of sub)
-            version_id = hasattr(targets[uid], 'version_id') and \
-                       targets[uid].version_id or None
-            if version_id is None:
-                # attempt initial save of unversioned targets
-                pr = getToolByName(instance, 'portal_repository')
-                if pr.isVersionable(targets[uid]):
-                    pr.save(obj=targets[uid],
-                            comment=to_utf8(ts(_("Initial revision"))))
+            version_id = getattr(targets[uid], 'version_id', None)
             if not hasattr(instance, 'reference_versions'):
                 instance.reference_versions = {}
-            if not hasattr(targets[uid], 'version_id'):
-                targets[uid].version_id = None
-            instance.reference_versions[uid] = targets[uid].version_id
+            instance.reference_versions[uid] = version_id
```

Comment:
Phew, that's kind of bad...  Do we really just set raw attribute on the object with a dict?
At least we should use here a PersitentDict, so that changes get signalled to the transaction machinery...

Anyhow, wouldn't it be possible in the `None` case to just implicitly say that it is the initial version instead of creating one during creation?

Reply:
Umm, yes you're right, but if we change this, then we will have to "port" this value for all Analysis in the upgrade step. HistoryAwareReferenceField` is only used in Analysis for `Calculation`, so although it presumably supports multivalued fields, I think that maybe we could just simply remove the "multivalue" support (and the `referencesSortable` thing as well) 
and just add a `CalculationVersion` StringField.... I recall that maybe we had this conversation already?

Reply:
> Anyhow, wouldn't it be possible in the `None` case to just implicitly say that it is the initial version instead of creating one during creation?

When you create an object for the first time, no version is created. The first version is created after you edit the object, when the version of the edited object is set to 0. Y realized then that the very first version of the object (after creation) was not accessible through version history anymore. This is the reason I added this last commit https://github.com/senaite/senaite.core/pull/1260/commits/5055e0e31b7e7b60b47f0d5c35e812e9cd5a04fe